### ISSUE TITLE
Non-nullable type is not narrowed when inside function

### ISSUE THREAD

**Issue by OliverJAsh** [Microsoft/TypeScript#20105]
**TypeScript Version:**  2.6.1

**Code**

```ts
declare const obj: {
    myFn: (() => number) | undefined;
};

// No errors
if (obj.myFn !== undefined) obj.myFn();

if (obj.myFn !== undefined) {
    const a = () =>
        // Unexpected error: Object is possibly 'undefined'
        obj.myFn();
    a();
}
```

**Comment by andy-ms** [comment]
Duplicate of #9998
